warehouse/internal/repo/staging.go

```diff
@@ -0,0 +1,268 @@
+package repo
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/timeutil"
+	"github.com/rudderlabs/rudder-server/warehouse"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+const tableName = warehouseutils.WarehouseStagingFilesTable
+
+const tableColumns = `
+	id,
+    location,
+    source_id,
+    destination_id,
+    schema,
+    error,
+    status,
+    first_event_at,
+    last_event_at,
+    total_events,
+    created_at,
+    updated_at,
+    metadata,
+    workspace_id 
+`
+
+type StagingFiles struct {
+	DB  *sql.DB
+	Now func() time.Time
+
+	once sync.Once
+}
+
+type metadataSchema struct {
+	UseRudderStorage      bool   `json:"use_rudder_storage"`
+	SourceBatchID         string `json:"source_batch_id"`
+	SourceTaskID          string `json:"source_task_id"`
+	SourceTaskRunID       string `json:"source_task_run_id"`
+	SourceJobID           string `json:"source_job_id"`
+	SourceJobRunID        string `json:"source_job_run_id"`
+	TimeWindowYear        int    `json:"time_window_year"`
+	TimeWindowMonth       int    `json:"time_window_month"`
+	TimeWindowDay         int    `json:"time_window_day"`
+	TimeWindowHour        int    `json:"time_window_hour"`
+	DestinationRevisionID string `json:"destination_revision_id"`
+}
+
+func metadataFromStagingFile(stagingFile warehouse.StagingFileT) metadataSchema {
```

Comment:
```suggestion
func metadataFromStagingFile(stagingFile *warehouse.StagingFileT) metadataSchema {
```
Should we use something like this?